src/Agent.Plugins/PipelineCache/SavePipelineCacheV0.cs

```diff
@@ -29,6 +29,14 @@ public class SavePipelineCacheV0 : PipelineCacheTaskPluginBase
             string salt,
             CancellationToken token)
         {
+            var JobStatus = context.Variables["agent.jobstatus"];
+
+            if(JobStatus != null && !JobStatus.Value.Equals("Succeeded"))
```

Comment:
I think if JobStatus is null then we should not save. Does Variables have a TryGet?

Reply:
i.e. only cache if variables has agent.jobstatus, it's associated value exists and the value is Succeeded

---
In reply to: [298396710](https://github.com/microsoft/azure-pipelines-agent/pull/2320#discussion_r298396710) [](ancestors = 298396710)

Reply:
Variables is a Dictionary. I guess we can have a check like, JobStatus = context.Variables.ContainsKey("agent.jobstatus"). If it's null, throw. If not, if Value == Succeeded ?

Reply:
Let's not throw. Let's just only save the cache when we're 100% sure it's safe.  It's ok to not save the caches sometimes when we should. It's not ok to ever save the cache when we shouldn;t

Reply:
Got it. Added the check to check whether key exists.

Reply:
I think [] indexing a dictionary will throw if the key doesn't exist